addons/point_of_sale/models/pos_config.py

```diff
@@ -26,10 +26,23 @@ def _default_invoice_journal(self):
         return self.env['account.journal'].search([('type', '=', 'sale'), ('company_id', '=', self.env.company.id)], limit=1)
 
     def _default_payment_methods(self):
-        return self.env['pos.payment.method'].search([('split_transactions', '=', False), ('company_id', '=', self.env.company.id)])
+        """ Should only default to payment methods that are compatible to this config's company and currency.
+        """
+        return self.env['pos.payment.method'].search([
+            ('split_transactions', '=', False),
+            ('company_id', 'in', (False, self.env.company.id)),
+            '|',
+                ('journal_id', '=', False),
+                ('journal_id.currency_id', '=', False),
```

Comment:
why not using journal_id.currency_id', 'in', (False, self.currency_id.id)

Reply:
BTW why do we use self.currency_id.id or self.env.company.currency_id.id as default is always called the same way I don't why "self.currency_id" could be either set or not set

Reply:
"self" can be the record being edited, e.g. when in "write".